packages/Heading/src/Heading.styles.js

```diff
@@ -64,5 +63,6 @@ export const headingStyles = props => `
   ${heading[`HEADING_LEVEL_${props.safeDisplayLevel || props.safeLevel}`]}
   ${props.isHidden ? stylers.visuallyHidden : ""}
   ${props.hasUnderline ? underline(props.safeDisplayLevel || props.safeLevel) : ""}
+  ${props.hasDivider ? "display: flex;" : ""}
```

Comment:
Most of the time it's fine if the `<Heading>` is a `flex` instead of a `block`, but it's irritating to override to to be a `block` when it does cause issues, since it's reasonable for app developers to expect an `<h>` element to be a `block`.